Dark Mode: My Store Tab - Misc

AmandaRiu opened the pull request:
Fixes #1767 by completing the following:
- Apply dark/light styling to `MyStatsStoreAvailabilityCard`
- Apply dark/light styling to `WCEmptyView`
- Apply dark/light styling to `MyStoreStatsRevertedNoticeCard` 
- Cleanup style definitions and associated layout files
- Add Card Expander styles
- Add Button styles

## Waiting for Customers
### API 21
![Screen Shot 2020-01-02 at 11 27 12 PM](https://user-images.githubusercontent.com/5810477/71710039-80b78f80-2db7-11ea-898f-af8aa918b5a1.png)
### API 28
![Screen Shot 2020-01-02 at 11 27 40 PM](https://user-images.githubusercontent.com/5810477/71710044-82815300-2db7-11ea-9a48-3d7d22bbac10.png)

## Try our improved stats
### API 21
![stats-avail-21-collapsed](https://user-images.githubusercontent.com/5810477/71710171-2834c200-2db8-11ea-84c4-e59c56569e07.png)
![stats-avail-21-expanded](https://user-images.githubusercontent.com/5810477/71710173-2a971c00-2db8-11ea-9292-298417d3975b.png)
### API 28
![stats-avail-28-collapsed](https://user-images.githubusercontent.com/5810477/71710177-308cfd00-2db8-11ea-9af2-fda31a960307.png)
![stats-avail-28-expanded](https://user-images.githubusercontent.com/5810477/71710180-3256c080-2db8-11ea-97b8-2fad45ac9347.png)

## Stats Reverted
### API 21
![stats-reverted-21](https://user-images.githubusercontent.com/5810477/71710359-1869ad80-2db9-11ea-927c-e1443ac33974.png)
### API 28
![stats-reverted-28](https://user-images.githubusercontent.com/5810477/71710363-1dc6f800-2db9-11ea-93f0-c776f7d54766.png)

Commit:
Add new button styles with custom color selector to override colors

Commit:
Style the Stats Reverted dashboard card

Commit:
Merge branch 'darkmode/1765-top-performers-v2' into darkmode/1767-my-store-fine-tuning

Commit:
Remove the Woo.Card.WithoutPadding style

Commit:
Convert from LinearLayout to MaterialCardView

Commit:
Code cleanup

Commit:
Add expandable card style and clean up resources

- Consolidate arrow-type resources and delete unused types
- Add card style to theme

Commit:
Add tools context for better preview in design view

Commit:
Style MyStoreStatsAvailablility card

Commit:
Convert to using ConstraintLayout

Allows for more predictable margin management

Commit:
Convert to MaterialCardView

Commit:
Theme and style the "No orders" view

Commit:
Remove unused FrameLayout layer

Commit:
Move base styles to styles_base.xml

Commit:
Misc code cleanup

AmandaRiu commented:
@Garance91540 Ready for a design review! 👍

Commit:
Merge branch 'darkmode/1765-top-performers-v2' into darkmode/1767-my-store-fine-tuning

Commit:
Merge branch 'darkmode/1765-top-performers-v2' into darkmode/1767-my-store-fine-tuning

Commit:
Swap out Button and TextView components with Material components

Garance91540 commented:
Hi @AmandaRiu , lots of progress here! 

Here are my few notes:
- Illustrations' colors need to be updated on dark mode to preserve contrast ratio. I updated all illustrations for all screens (I realize I haven't add them to the Figma file, will try to get to it today), but I didn't update this one since we are changing this screen with the new empty states. 

<img width="479" alt="Screen Shot 2020-01-06 at 12 36 01 PM" src="https://user-images.githubusercontent.com/28450084/71815875-1e91af80-3081-11ea-9552-527e11bcbe0e.png">

Is it possible to add this new illustrations and update the light mode too?

- Fonts appear slightly smaller on API 28, is it possible to make them the same size as API 21?
- On the screenshots you shared for stats reverted, it seems like the vertical padding between the "Visitors", "Orders", "Revenue" title and the date got removed. 
<img width="294" alt="Screen Shot 2020-01-06 at 12 43 27 PM" src="https://user-images.githubusercontent.com/28450084/71816255-4cc3bf00-3082-11ea-9099-b455085ca97f.png">
<img width="290" alt="Screen Shot 2020-01-06 at 12 43 21 PM" src="https://user-images.githubusercontent.com/28450084/71816258-4fbeaf80-3082-11ea-94d4-f1a533d47d50.png">

- Is it possible to update the notification dot color for the "try new stats" banner to Pink 50? I'm seeing it black on my end (and on your screenshots for API 28

thanks! Let me know if you have any questions

Commit:
Add margin above main stats row

Commit:
Remove tint since we need the dot to be a different color

AmandaRiu commented:
Thanks for the review @Garance91540!

> Is it possible to add these new illustrations and update the light mode too? I've added them to our Figma file: https://www.figma.com/file/S62xbAwtgoMgax4GcuFjKS/System---Dark%2FLight-modes?node-id=0%3A1

I'd prefer to make those changes in a separate project for empty states - otherwise it's possible this feature may never get finished since it's already massive. Right now my focus is only to convert the components used to material components, and then to apply light/dark mode styling. 

> Fonts appear slightly smaller on API 28, is it possible to make them the same size as API 21?

this is only because the device I'm running API 28 is never and has a MUCH higher resolution. It doesn't have anything to do with the version and should be expected. 

> On the screenshots you shared for stats reverted, it seems like the vertical padding between the "Visitors", "Orders", "Revenue" title and the date got removed.

> On the screenshots you shared for stats reverted, it seems like the vertical padding between the "Visitors", "Orders", "Revenue" title and the date got removed.
 
Oh no! Fixed in beb57018db4e8c990c2593a58abd21977a57d8dd. Here's what it looks like now:

![Screen Shot 2020-01-06 at 9 42 44 AM](https://user-images.githubusercontent.com/5810477/71832865-07df5e80-3069-11ea-9fdf-b3beb7b18283.png)

> Is it possible to update the notification dot color for the "try new stats" banner to Pink 50? I'm seeing it black on my end (and on your screenshots for API 28

Great catch! Fixed in 7f759b531977c0e2677aa3489e91544283ad9b25. Here's what it looks like now:

![Screen Shot 2020-01-06 at 9 49 13 AM](https://user-images.githubusercontent.com/5810477/71833333-ee8ae200-3069-11ea-9959-0041df1e4563.png)

Ready for another round! 👍

Commit:
Convert to using a material button where available and update style

Commit:
Style the colored buttons using  the secondary instead of primary

The material theme automatically chooses colorPrimary for the background
color of colored buttons, but the designs call for the secondary color
to be used.

Commit:
Merge branch 'feature/dark-mode' into darkmode/1767-my-store-fine-tuning

AmandaRiu commented:
This PR is ready for review. No need to wait for design review, I can address design comments in a different PR. Thanks!

nbradbury reviewed (approved):
Looks great! :shipit:

kyleaparker commented:
👍
